src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/HotReload/ProjectHotReloadSessionManager.cs

```diff
@@ -0,0 +1,233 @@
+﻿// Licensed to the .NET Foundation under one or more agreements. The .NET Foundation licenses this file to you under the MIT license. See the LICENSE.md file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.ComponentModel.Composition;
+using System.Diagnostics;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.VisualStudio.HotReload.Components.DeltaApplier;
+using Microsoft.VisualStudio.ProjectSystem.Debug;
+using Microsoft.VisualStudio.ProjectSystem.Properties;
+using Microsoft.VisualStudio.Threading;
+
+namespace Microsoft.VisualStudio.ProjectSystem.VS.HotReload
+{
+    [Export(typeof(IProjectHotReloadSessionManager))]
+    internal class ProjectHotReloadSessionManager : IProjectHotReloadSessionManager
+    {
+        private readonly UnconfiguredProject _project;
+        private readonly IActiveDebugFrameworkServices _activeDebugFrameworkServices;
+        private readonly Lazy<IProjectHotReloadAgent> _projectHotReloadAgent;
+        private readonly Lazy<IHotReloadDiagnosticOutputService> _hotReloadDiagnosticOutputService;
+
+        private HotReloadState? _pendingSessionState = null;
+
+        private ImmutableDictionary<int, HotReloadState> _activeSessions = ImmutableDictionary<int, HotReloadState>.Empty;
+
+        private int _nextUniqueId = 1;
+
+        [ImportingConstructor]
+        public ProjectHotReloadSessionManager(
+            UnconfiguredProject project,
+            IActiveDebugFrameworkServices activeDebugFrameworkServices,
+            Lazy<IProjectHotReloadAgent> projectHotReloadAgent,
+            Lazy<IHotReloadDiagnosticOutputService> hotReloadDiagnosticOutputService)
+        {
+            _project = project;
+            _activeDebugFrameworkServices = activeDebugFrameworkServices;
+            _projectHotReloadAgent = projectHotReloadAgent;
+            _hotReloadDiagnosticOutputService = hotReloadDiagnosticOutputService;
+        }
+
+        public async Task ActivateSessionAsync(int processId)
+        {
+            if (_pendingSessionState is not null)
+            {
+                Assumes.NotNull(_pendingSessionState.Session);
+
+                try
+                {
+                    Process? process = Process.GetProcessById(processId);
+
+                    await _hotReloadDiagnosticOutputService.Value.WriteLineAsync($"{_pendingSessionState.Session.Name}: Attaching to process '{processId}'.");
```

Comment:
This diagnostic output is only for our (Microsoft) purposes, yes? Not really for the customer?

Reply:
It is primarily for development purposes at this point. What we want to display to the customer is TBD and also covers a bunch of Hot Reload components we don't own.

It all goes to the "Hot Reload" pane of the Output Window.

Reply:
Ah, okay. So... since it is a 'log', I'm assuming we don't do translations on log information?

Reply:
If these end up being hidden behind a flag or reg key or some such we don't need to translate them. If they are on by default (e.g. the current behavior) we will need to translate them because they are still user-facing. I'll probably pull them out into string resources because there's no real reason not to.